lib/server.js

```diff
@@ -19,30 +19,50 @@ util.buildIndex(isDist); // throw compilation errors right away
 let app = express();
 let env = util.readEnv(isDist);
 let port = parseInt(process.env.PORT || 4300);
-app.use(morgan('combined', { skip: req => !util.isIndex(req.path) }));
+app.use(morgan('combined', {
+  skip: req => req.path === '/' || !util.isIndex(req.path)
+}));
 
 // proxy feed requests
-app.get('/proxy', function proxyFeed(req, res) {
-  res.setHeader('Cache-Control', 'public, max-age=30');
-
-  // monkey patch
-  let feedUrl = decodeURIComponent(req.query.url);
-  feedUrl = feedUrl.replace(/^https:\/\/feeds\.earhustlesq\.com/, 'http://feeds.earhustlesq.com');
-
-  let parts = url.parse(feedUrl);
-  let requester = (parts.protocol === 'http:' ? http : https);
-  let proxyReq = requester.request(parts.href, (proxyRes) => {
-    proxyRes.setEncoding('utf8');
-    proxyRes.on('data', (chunk) => res.write(chunk));
-    proxyRes.on('close', () => res.end());
-    proxyRes.on('end', () => res.end());
-  }).on('error', (e) => {
-    console.log(e.message);
-    res.writeHead(500);
-    res.end();
-  });
-  proxyReq.end();
+let xmlMimes = [
+  'application/rss+xml',
+  'application/rdf+xml;q=0.8',
+  'application/atom+xml;q=0.6',
+  'application/xml;q=0.4',
+  'text/xml;q=0.4',
+].join(',');
+app.get('/proxy', (req, res, next) => {
+  let parts = url.parse(req.query.url || '');
+  if (parts.protocol && parts.host) {
+    req.proxyUrl = `${parts.protocol}//${parts.host}${parts.path || ''}`;
+    next();
+  } else {
+    console.error(`Proxy error: invalid url '${req.url}'`);
+    res.status(400).send('Invalid proxy url');
+  }
 });
+app.get('/proxy', proxy({
+  target: 'http://dummy-proxy-host.prx.org',
+  router: req => req.proxyUrl,
+  onProxyReq: req => {
+    req.setHeader('accept', xmlMimes);
+    req.setHeader('user-agent', 'play.prx.org feed proxy');
```

Comment:
These are changes - previously we were making a header-less request.  This proxy lib adds all the original-request headers by default, so I have to override a few to make feedburner return XML not HTML.

Feedburner (which hosts most of our feeds) ignores the `Accept` header.  And has some strange regexps it applies to the `Referer` and `User-Agent` headers to decide if it should return HTML.  Changing the `User-Agent` to something non-browser-ish guarantees we get XML back.